ocs_ci/ocs/resources/mcg.py

```diff
@@ -183,7 +183,10 @@ def __init__(self, *args, **kwargs):
                     not pods
                 ), "RGW pods should not exist in the current platform/cluster"
 
-        elif config.ENV_DATA.get("platform") in constants.ON_PREM_PLATFORMS:
+        elif (
+            config.ENV_DATA.get("platform") in constants.ON_PREM_PLATFORMS
+            and not config.ENV_DATA["mcg_only_deployment"]
+        ):
```

Comment:
should be checked in if lines 171-174 same as in [here](https://github.com/red-hat-storage/ocs-ci/pull/5120/files#diff-1d32a061b0f98ee283c52d1f7e9ca822177982cee47f7daa76904f5d8e1c3bfdv)

Reply:
we can't use lines 171-174 same as in here, since its not TC

Reply:
in the if statement above there is a check to verify that there are no rgw related pods in the deployment (mcg-only shouldnt have any) and the addition to this if statement skips the whole check

Reply:
@bernerhat I didn't see any addition checks in this block . Only rgw verification is there in this block which we are skipping for MCG Only mode

Reply:
Why would lines 171-174 be limited to a test case?
The current flow of the PR will mean we're not verifying that RGW pods are missing in MCG-only (which certainly doesn't hurt).
If we'll change the if in these lines, it'll mean we'll *always* verify RGW pods are missing in MCG-only deployments -
```
if (
  config.ENV_DATA["platform"].lower() in constants.CLOUD_PLATFORMS
   or storagecluster_independent_check()
   or config.ENV_DATA["mcg_only_deployment"]
):
```
It's essentially better coverage for free

Reply:
@MeridianExplorer I think we can RGW pods missing checks in post install verification of MCG only deployment instead of here, so that we have all checks at one place. I will take it up in another PR.